dbt_invoke/properties.py

```diff
@@ -258,27 +258,35 @@ def _transform_ls_results(ctx, **kwargs):
     """
     # Run dbt ls to retrieve resource path and json information
     _LOGGER.info('Searching for matching resources...')
-    result_lines_path = _utils.dbt_ls(
-        ctx,
-        supported_resource_types=_SUPPORTED_RESOURCE_TYPES,
-        logger=_LOGGER,
-        **kwargs,
-    )
-    result_lines_dict = _utils.dbt_ls(
+    potential_results = _utils.dbt_ls(
         ctx,
         supported_resource_types=_SUPPORTED_RESOURCE_TYPES,
         logger=_LOGGER,
         output='json',
         **kwargs,
     )
-    results = dict(zip(result_lines_path, result_lines_dict))
-    # Filter dictionary for existing files and supported resource types
-    results = {
-        k: v
-        for k, v in results.items()
-        if v['resource_type'] in _SUPPORTED_RESOURCE_TYPES
-        and Path(ctx.config['project_path'], k).exists()
-    }
+    results = dict()
+    for i, potential_result in enumerate(potential_results):
+        if 'original_file_path' in potential_result:
+            potential_result_path = potential_result['original_file_path']
+        # Before dbt version 0.20.0, original_file_path was not
+        # included in the json response of "dbt ls". For older
+        # versions of dbt, we need to run "dbt ls" with the
+        # "--output path" argument in order to retrieve paths
+        else:
+            potential_result_paths = _utils.dbt_ls(
+                ctx,
+                supported_resource_types=_SUPPORTED_RESOURCE_TYPES,
+                logger=_LOGGER,
+                output='path',
+                **kwargs,
+            )
+            assert len(potential_result_paths) == len(
+                potential_results
+            ), 'Length of results differs from length of result details'
```

Comment:
If potential_results contains multiple items without original_file_path, does that mean this section will run multiple times with the same result each time? What is the downside of moving this section outside of the for loop entirely so it runs once no matter what? 

I understand that if the user has a version of dbt > 0.20.0, the section is unnecessary, but potentially having it run multiple times for older versions is also odd to me.

Reply:
Good catch!  I'd like to run it zero times when "original_file_path" exists, but only once total when "original_file_path" is not present for any of the `potential_results`.  I'll make the necessary changes.